src/toil/leader.py

```diff
@@ -77,8 +77,13 @@ def callback(fileHandle):
             except AttributeError:
                 pass
             else:
-                for log in logs:
-                    logger.info("%s:    %s", log.jobStoreID, log.text)
+                jobStoreIDs = {log.jobStoreID for log in logs}
+                # we may have multiple jobs per worker
+                for jobStoreID in jobStoreIDs:
+                    jobLogs = [log.text for log in filter(lambda log: log.jobStoreID == jobStoreID, logs)]
```

Comment:
Since you consume the list immediately, this could be generator expression. Making it guarded would also allow you to get rid of filter and lambda.